client/login/desktop-login/index.tsx

```diff
@@ -0,0 +1,115 @@
+import './style.scss';
+import { Button } from '@wordpress/components';
+import debugFactory from 'debug';
+import { useTranslate } from 'i18n-calypso';
+import { useEffect, useState } from 'react';
+import FormattedHeader from 'calypso/components/formatted-header';
+import Main from 'calypso/components/main';
+import { Notice } from 'calypso/components/notice';
+import WpcomLoginForm from 'calypso/signup/wpcom-login-form';
+
+const debug = debugFactory( 'calypso:desktop' );
+
+// The desktop app will intercept this URL and start the login in the user's external browser.
+const loginUrl = '/desktop-start-login';
+
+type Props =
+	| { action: 'start'; accessToken?: never; error?: never }
+	| { action: 'finalize'; accessToken: string; error?: string };
+
+/**
+ * The login page of the WordPress.com Desktop app.
+ *
+ * Initially (action=start), a button is rendered that when clicked sends the user
+ * to their browser (outside the desktop app), so that they can log in to WordPress.com.
+ *
+ * When authentication is complete, the user is redirected back to the desktop app
+ * and ends up here (action=finalize), with their access token passed as a prop.
+ *
+ * We then use that access token to submit the login form, which will set the cookie,
+ * and thus log the user in.
+ */
+export default function DesktopLogin( props: Props ) {
```

Comment:
Do the `start` and `finalize` versions of the component have anything in common? Except the `<Main>` wrapper and the error notice?

Seems to me we'd be better off with two components, `DesktopLoginStart` and `DesktopLoginFinalize`. Currently we're merging two components together, with a lot of conditional logic inside.

Reply:
I've pushed a commit that splits it into two components.